File: output_repair/claude-opus/sample_1/repair_1/c/c/Struct/account.c
#include "stdlib.h"

struct account
{
    int balance;
};

/*@ predicate account(struct account *a) = a->balance |-> _; @*/

struct account *create_account()
//@ requires true;
//@ ensures account(result) &*& result != 0;
{
    struct account *myAccount = malloc(sizeof(struct account));
    if (myAccount == 0)
    {
        abort();
    }
    myAccount->balance = 0;
    return myAccount;
}

void account_set_balance(struct account *myAccount, int newBalance)
//@ requires account(myAccount);
//@ ensures account(myAccount);
{
    myAccount->balance = newBalance;
}

void account_dispose(struct account *myAccount)
//@ requires account(myAccount);
//@ ensures true;
{
    free(myAccount);
}

int main()
//@ requires true;
//@ ensures true;
{
    struct account *myAccount = create_account();
    account_set_balance(myAccount, 5);
    account_dispose(myAccount);
    return 0;
}

File: output_20260318_123118/deepseek/rust/Struct/account.rs
use std::alloc::{Layout, alloc, handle_alloc_error, dealloc};

struct Account {
    balance: i32,
}

/*@
predicate account_own(account: *mut Account; balance: i32) =
    struct_Account_padding(account) &*&
    (*account).balance |-> balance;
@*/

impl Account {
    #[requires(Layout::new::<Account>().size() > 0)]
    #[ensures(result != 0 as *mut Account)]
    #[ensures(account_own(result, 0))]
    unsafe fn create() -> *mut Account {
        let my_account = alloc(Layout::new::<Account>()) as *mut Account;
        if my_account.is_null() {
            handle_alloc_error(Layout::new::<Account>());
        }
        (*my_account).balance = 0;
        my_account
    }

    #[requires(account_own(my_account, _))]
    #[ensures(account_own(my_account, new_balance))]
    unsafe fn set_balance(my_account: *mut Account, new_balance: i32) {
        (*my_account).balance = new_balance;
    }

    #[requires(account_own(my_account, _))]
    #[ensures(true)]
    unsafe fn dispose(my_account: *mut Account) {
        dealloc(my_account as *mut u8, Layout::new::<Account>());
    }
}

fn main() {
    unsafe {
        let my_account = Account::create();
        Account::set_balance(my_account, 5);
        Account::dispose(my_account);
    }
}